libclc/ptx-nvidiacl/libspirv/images/image.cl

```diff
@@ -2270,6 +2272,249 @@ _CLC_DEFINE_SAMPLEDIMAGE_BINDLESS_READ_BUILTIN(half4, 1, Dv4_DF16_, v4f16, f, fl
+float4 __nvvm_tex_1d_v4f32_i32(unsigned long,
+                               int) __asm("__clc_llvm_nvvm_tex_1d_v4f32_s32");
+float4 __nvvm_tex_2d_v4f32_i32(unsigned long, int,
+                               int) __asm("__clc_llvm_nvvm_tex_2d_v4f32_s32");
+float4 __nvvm_tex_3d_v4f32_i32(unsigned long, int, int,
+                               int) __asm("__clc_llvm_nvvm_tex_3d_v4f32_s32");
+
+// Macro to generate texture vec4 fetches
+#define _CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(                  \
+    elem_t, fetch_elem_t, dimension, vec_size, fetch_vec_size, coord_input,    \
+    coord_parameter)                                                           \
+  elem_t##4 __nvvm_tex_##dimension##d_##vec_size##_i32(                        \
+      unsigned long imageHandle, coord_input) {                                \
+    fetch_elem_t##4 a = __nvvm_tex_##dimension##d_##fetch_vec_size##_i32(      \
+        imageHandle, coord_parameter);                                         \
+    return cast_##fetch_elem_t##4_to_##elem_t##4(a);                           \
+  }
+
+// Macro to generate texture vec2 fetches
+#define _CLC_DEFINE_BINDLESS_VEC2THUNK_TEXTURE_FETCH_BUILTIN(                  \
+    elem_t, fetch_elem_t, dimension, vec_size, fetch_vec_size, coord_input,    \
+    coord_parameter)                                                           \
+  elem_t##2 __nvvm_tex_##dimension##d_##vec_size##_i32(                        \
+      unsigned long imageHandle, coord_input) {                                \
+    fetch_elem_t##4 a = __nvvm_tex_##dimension##d_##fetch_vec_size##_i32(      \
+        imageHandle, coord_parameter);                                         \
+    return cast_##fetch_elem_t##4_to_##elem_t##2(a);                           \
+  }
+
+// Macro to generate texture singular data type fetches
+#define _CLC_DEFINE_BINDLESS_THUNK_TEXTURE_FETCH_BUILTIN(                      \
+    elem_t, fetch_elem_t, dimension, vec_size, fetch_vec_size, coord_input,    \
+    coord_parameter)                                                           \
+  elem_t __nvvm_tex_##dimension##d_##vec_size##_i32(unsigned long imageHandle, \
+                                                    coord_input) {             \
+    return (elem_t)__nvvm_tex_##dimension##d_##fetch_vec_size##_i32(           \
+        imageHandle, coord_parameter)[0];                                      \
+  }
+
+#define COORD_INPUT_1D int x
+#define COORD_INPUT_2D int x, int y
+#define COORD_INPUT_3D int x, int y, int z
+
+#define COORD_PARAMS_1D x
+#define COORD_PARAMS_2D x, y
+#define COORD_PARAMS_3D x, y, z
+
+_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(short, int, 1, v4i16, v4i32, COORD_INPUT_1D, COORD_PARAMS_1D)
+_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(short, int, 2, v4i16, v4i32, COORD_INPUT_2D, COORD_PARAMS_2D)
+_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(short, int, 3, v4i16, v4i32, COORD_INPUT_3D, COORD_PARAMS_3D)
```

Comment:
Nit: We could reduce code size a little more here if we consolidate these a little through another macro:
```suggestion
#define _CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN_ALL_DIMS(elem_t, fetch_elem_t, vec_size, fetch_vec_size) \
_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(elem_t, fetch_elem_t, 1, vec_size, fetch_vec_size, COORD_INPUT_1D, COORD_PARAMS_1D) \
_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(elem_t, fetch_elem_t, 2, vec_size, fetch_vec_size, COORD_INPUT_2D, COORD_PARAMS_2D) \
_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN(elem_t, fetch_elem_t, 3, vec_size, fetch_vec_size, COORD_INPUT_3D, COORD_PARAMS_3D) \

_CLC_DEFINE_BINDLESS_VEC4THUNK_TEXTURE_FETCH_BUILTIN_ALL_DIMS(short, int, v4i16, v4i32)
```
Seem to me like that would apply to most, if not all, of the following definitions. Just a preference given how big the file is.